File: 2023/day01-1.hs
import System.IO
import Data.Char

solution :: String -> Int
solution content = foldr (+) 0 . map parse . lines $  content

parse :: String -> Int
parse xs = 10 * (firstDigit xs) + (firstDigit (reverse xs))

firstDigit :: String -> Int
firstDigit (x:xs)
    | x `elem` "1234567890" = digitToInt x
    | otherwise = firstDigit xs

main :: IO ()
main = do
    handle <- openFile "day01.dat" ReadMode
    content <- hGetContents handle
    print . solution $ content

File: 2023/day01-1.ts
function solution(content: string) {
  return content
    .split("\n")
    .map(
      (line) =>
        firstDigit(line) * 10 + firstDigit(line.split("").reverse().join(""))
    )
    .reduce((acc, cur) => acc + cur, 0);
}

function firstDigit(line: string) {
  return parseInt(
    line.split("").find((ch) => "1234567890".includes(ch)) ?? "-1"
  );
}

Deno.readTextFile("day01.dat")
  .then((content) => content.trim())
  .then(solution)
  .then(console.log);
